Add recipe for ligature-pragmatapro

wavexx opened the pull request:
### Brief summary of what the package does

This package provides pre-baked [ligature](https://github.com/mickeynp/ligature.el) configuration for the [PragmataPro](https://fsd.it/shop/fonts/pragmatapro/) programming font.

### Direct link to the package repository

https://gitlab.com/wavexx/ligature-pragmatapro.el

### Your association with the package

Author

### Relevant communications with the upstream package maintainer

n/a

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

- You're dependent on a transitive import of `rx`, but just to be safe you should be explicit and use `(eval-when-compile (require 'rx))`
- Double-check the regular expressions -- virtually all of your single backslashes are not going to appear in the resulting regular expression, e.g. `"\+"` is equivalent to `"+"` as far as Emacs is concerned, and instead you probably want `\\+` or `[+]`.  Emacs lisp mode should highlight these using `font-lock-warning-face` in most cases, so look for color changes in the regexps

This also sort of feels like a piece of configuration more than it does a package, although I think there's value in packaging up this list of regular expressions.  But I feel like you should wrap `(ligature-set-ligatures 'prog-mode ligature-pragmatapro-ligatures)` up in a function call (and tell the user to call e.g. `(ligature-pragmatapro-setup)` in their config), so the user knows exactly what's happening -- I discourage global side effects like these at import time.

wavexx commented:
>     * You're dependent on a transitive import of `rx`, but just to be safe you should be explicit and use `(eval-when-compile (require 'rx))`

Done.

>     * Double-check the regular expressions -- virtually all of your single backslashes are not going to appear in the resulting regular expression, e.g. `"\+"` is equivalent to `"+"` as far as Emacs is concerned, and instead you probably want `\\+` or `[+]`.  Emacs lisp mode should highlight these using `font-lock-warning-face` in most cases, so look for color changes in the regexps

You're right. I produced this through some passes of sed, so they're mostly superfluous. Cleaned them up.

> This also sort of feels like a piece of configuration more than it does a package, although I think there's value in packaging up this list of regular expressions. But I feel like you should wrap `(ligature-set-ligatures 'prog-mode ligature-pragmatapro-ligatures)` up in a function call (and tell the user to call e.g. `(ligature-pragmatapro-setup)` in their config), so the user knows exactly what's happening -- I discourage global side effects like these at import time.

It is absolutely pure configuration, in fact if you're importing the package it's because you're using this font, and that's the bare minimum you'll want to do. Since `ligature` does this already for the other fonts such as FiraCode, I just copied the behavior to avoid having useless mappings in the table.

wavexx commented:
I went ahead and modified the behavior (I also dislike load side-effects). I now perform the setup in `ligature-pragmatapro-setup`. Along with autoload cookies, this keeps the configuration also to one line.

I updated the instructions accordingly.

riscy commented:
Thanks so much for accommodating -- merging!
